### PR TITLE
Enforce maximum amount of users for new users only

### PR DESCRIPTION
**Pull request by jbtbnl**
When the maximum amount of users is reached, verdaccio-htpassword returned a 403 error when existing users tried to login. This PR is supposed to fix that.

Please verify/test before merging, I only performed a manual test (and concluded that this fixes the outlined issue).

### CODE REVIEW

**Comment by juanpicado** [comment]
What do you think @ayusharma  ?

**Comment by jbtbnl** [comment]
@juanpicado something like this? I'm a bit out of my comfort zone here as a PHP developer to be honest 😄

**Comment by juanpicado** [comment]
@jbtbnl I think would be rather something like this 

```
it('should throw error for existing username and password with max number of users reached', () => {
    const verifyFn = jest.fn(() => true);
    const input = sanityCheck('testNew', users.test, verifyFn, users, 1);

    expect(input.status).toEqual(403);
    expect(input.message).toEqual('maximum amount of users reached');
    expect(verifyFn).toHaveBeenCalledTimes(0);
  });
```

At the end what you are trying to test is **'maximum amount of users reached'**. Could you ammend it?

**Comment by jbtbnl** [comment]
@juanpicado the behaviour I'm testing here is that error 403 (maximum amount of users reached) should *not* be thrown when an existing user tries to login (with valid credentials). Instead, NPM client seems to be happy with a 409 error (username is already registered) because the login is successful then.

My test breaks on verdaccio-htpasswd@master, while it passes given the changes in this PR.
Your test passes on both versions. While the test might be valid, it seems to be unrelated to this PR.

File: src/utils.js [hunk 1]
```diff
@@ -144,6 +137,11 @@ export function sanityCheck(
     // $FlowFixMe
     err.status = 401;
     return err;
+  } else if (Object.keys(users).length >= maxUsers) {
```
Comment (ayusharma) [inline src/utils.js]: Also, I noticed one thing. We have `-1` to disable registration. In that case we have to check that too. 
```
. . .
if(maxUsers !== -1 && Object.keys(users).length >= maxUsers) {
. . .
```

Comment (juanpicado) [reply src/utils.js]: @ayusharma I think `Object.keys(users).length >= maxUsers` already handle that. It's a bit sublte but still, returns `409`

**Comment by juanpicado** [comment]
@jbtbnl it has been merged on verdaccio master. All seems to works fine. All test passes

**Pull request merged**